File: src/Goods.Tools/Types/Page.cs
namespace Goods.Tools.Types;

public readonly struct Page<T>(T[] values, Int32 totalRows)
{
	public T[] Values { get; } = values;
	public Int32 TotalRows { get; } = totalRows;

    public Page<TConverted> Convert<TConverted>(Func<T, TConverted> converter)
    {
        return new Page<TConverted>(Values.Select(converter).ToArray(), TotalRows);
    }
}

File: src/Students.BackOffice/client-app/src/tools/types/page.ts
export class Page<T> {
	constructor(public values: T[], public totalRows: number) {}

	static get default() {
		return new Page([], 0);
	}

	static convert<T>(data: any, func: (any: any) => T) {
		return new Page<T>((data.values as any[]).map(func), data.totalRows);
	}
}
